### DISCUSSION TITLE
Possible to turn off CPU starvation checker

### DISCUSSION THREAD

**Discussion by hamnis** [typelevel/cats-effect#3282] [Q&A] [answered]
Hello.

Is it possible to turn off the CPU starvation checker?
We have an app at $WORK that is not CPU starved, but uses other threadpools than the cats-effect one.
Since the CPU is saturated, we constantly get this message after cats-effect 3.4:

```
[WARNING] Your CPU is probably starving. Consider increasing the granularity
of your delays or adding more cedes. This may also be a sign that you are
unintentionally running blocking I/O operations (such as File or InetAddress)
without the blocking combinator.
```

**Comment by armanbilge** [comment, marked as answer]
Yes, set the `cpuStarvationCheckInitialDelay` to `Duration.Inf`
https://github.com/typelevel/cats-effect/blob/0c6abf384cd74f37d109a759f5563d32afc8ec02/core/shared/src/main/scala/cats/effect/unsafe/IORuntimeConfig.scala#L30

**Reply by hamnis** [reply]
Perfect. Thank you.

**Comment by djspiewak** [comment]
> We have an app at $WORK that is not CPU starved, but uses other threadpools than the cats-effect one.
> Since the CPU is saturated, we constantly get this message after cats-effect 3.4:

Fwiw, this probably means your CPU is starved. :-) At least, if we define "starvation" as in terms of the amount of time it would take an external async completion (such as a new incoming connection) to actually get scheduled and recognized. There's a lot of assumptions baked into that definition though, which might not apply to your app.

Definitions aside, the fun thing about the starvation checker is it indirectly detects starvation caused by contention from threads that Cats Effect doesn't control. So the external pool usage doesn't really have an impact here, it's just a measurement of whether or not the synthesis of the Cats Effect and the OS kernel schedulers are able to keep up with interrupts in a timely fashion.

**Comment by cquiroz** [comment]
I've been asked about this message from our users (of a large scala.js app) and thanks for the workaround.
While I understand the benefits of this warning it is fairly difficult to pintpoint where the problem is with the current message, especially in scala.js (we obviously don't do file I/O and all remote calls are done with fetch)
Would it be possible to somehow indicate what task are triggering the warning?

**Reply by djspiewak** [reply]
I'm going to write up some stuff for the site on this over the next week or so. The short answer is you can turn it off by setting the check interval to `Inf`. Pinpointing the task isn't possible in general because it might not be a task that is at fault: it might be your hosting environment! There are many possible causes and debugging steps. I'll try to clarify what I would recommend to track it down.
